bodo/io/iceberg/read_compilation.py

```diff
@@ -236,3 +209,106 @@ def determine_str_as_dict_columns(
         if metric < bodo.io.parquet_pio.READ_STR_AS_DICT_THRESHOLD:
             str_as_dict.add(str_col_names_to_check[i])
     return str_as_dict
+
+
+def get_iceberg_orig_schema(
+    conn_str: str,
+    table_id: str,
+    is_merge_into_cow: bool = False,
+) -> tuple[list[str], list[types.ArrayCompatible], pa.Schema]:
+    """
+    Helper function to fetch Bodo types for an Iceberg table with the given
+    connection info. Will include an additional Row ID column for MERGE INTO
+    COW operations.
+
+    Returns:
+        - List of column names
+        - List of column Bodo types
+        - PyArrow Schema Object
+    """
+    _ = verify_pyiceberg_installed()
+
+    # Get the pyarrow schema from Iceberg
+    catalog = conn_str_to_catalog(conn_str)
+    table = catalog.load_table(table_id)
+    return _get_table_schema(table, is_merge_into_cow)
+
+
+def get_orig_and_runtime_schema(
+    conn_str: str,
+    table_id: str,
+    selected_cols: list[str] | None = None,
+    read_as_dict_cols: list[str] = EMPTY_LIST,
+    detect_dict_cols: bool = False,
+    is_merge_into_cow: bool = False,
+) -> tuple[
+    # Original Schemas
+    list[str],
+    list[types.ArrayCompatible],
+    pa.Schema,
+    # Runtime Schema
+    list[str],
+    list[types.ArrayCompatible],
```

Comment:
I messed up this part initially which caused some schema evolution tests to fail. Basically, this function should return the original schema info (names, types, PyArrow schema) and schema after apply BodoSQL column pruning.